Mathlib/CategoryTheory/SmallObject/Iteration/UniqueHom.lean

```diff
@@ -136,8 +134,114 @@ noncomputable def mkOfSucc
+      mkOfLimitNatTransApp φ hj j (by rfl) =
+      mkOfLimitNatTransApp φ hj i hi.le ≫ iter₂.F.map (homOfLE (by simpa using hi.le)) := by
+  rw [mkOfLimitNatTransApp_eq_of_lt φ hj i hi, mkOfLimitNatTransApp, dif_neg (by simp)]
+  exact (iter₁.isColimit j hj (by rfl)).fac _ ⟨i, hi⟩
+
+/-- Auxiliary definition for `mkOfLimit`. -/
+@[simps]
+def mkOfLimitNatTrans : iter₁.F ⟶ iter₂.F where
+  app := fun ⟨k, hk⟩ => mkOfLimitNatTransApp φ hj k hk
+  naturality := fun ⟨k₁, hk₁⟩ ⟨k₂, hk₂⟩ f => by
+    have hk : k₁ ≤ k₂ := leOfHom f
+    obtain h₂ | rfl := hk₂.lt_or_eq
+    · dsimp
+      rw [mkOfLimitNatTransApp_eq_of_lt _ hj k₂ h₂,
+        mkOfLimitNatTransApp_eq_of_lt _ hj k₁ (lt_of_le_of_lt hk h₂),
+        congr_app (φ k₁ (lt_of_le_of_lt hk h₂)) ((truncFunctor ε hk).map (φ k₂ h₂))]
+      exact natTrans_naturality (φ k₂ h₂) k₁ k₂ hk (by rfl)
+    · obtain h₁ | rfl := hk₁.lt_or_eq
+      · exact mkOfLimitNatTransApp_naturality_top _ hj _ h₁
+      · obtain rfl : f = 𝟙 _ := rfl
+        simp only [map_id, id_comp, comp_id]
+
+/-- The (unique) morphism between two objects in `Iteration ε j` when `j` is a limit element,
+assuming we have a morphism between the truncations to `Iteration ε i` for all `i < j`. -/
+def mkOfLimit : iter₁ ⟶ iter₂ where
+  natTrans := mkOfLimitNatTrans φ hj
+  natTrans_app_zero := by
+    simp [mkOfLimitNatTransApp_eq_of_lt φ _ ⊥ (by simpa only [bot_lt_iff_ne_bot] using hj.ne_bot)]
+  natTrans_app_succ i h := by
+    dsimp
+    have h' := hj.succ_lt h
+    rw [mkOfLimitNatTransApp_eq_of_lt φ hj _ h',
+      (φ _ h').natTrans_app_succ i (Order.lt_succ_of_not_isMax (not_isMax_of_lt h)),
+      mkOfLimitNatTransApp_eq_of_lt φ _ _ h,
+      congr_app (φ i h) ((truncFunctor ε (Order.le_succ i)).map (φ (Order.succ i) h'))]
+    dsimp
+
+end
+
+variable [WellFoundedLT J]
+
+instance : Nonempty (iter₁ ⟶ iter₂) := by
+  let P := fun (i : J) => ∀ (hi : i ≤ j),
+    Nonempty ((truncFunctor ε hi).obj iter₁ ⟶ (truncFunctor ε hi).obj iter₂)
+  suffices ∀ i, P i from this j (by rfl)
+  intro i
+  induction i using SuccOrder.limitRecOn with
+  | hm i hi =>
+    obtain rfl : i = ⊥ := by simpa using hi
+    exact fun hi' ↦ ⟨Hom.mkOfBot _ _⟩
+  | hs i hi hi' =>
+    exact fun hi'' ↦ ⟨Hom.mkOfSucc _ _ hi (hi' ((Order.le_succ i).trans hi'')).some⟩
+  | hl i hi hi' =>
+    exact fun hi'' ↦ ⟨Hom.mkOfLimit (fun k hk ↦ (hi' k hk (hk.le.trans hi'')).some) hi⟩
+
+noncomputable instance : Unique (iter₁ ⟶ iter₂) :=
+  uniqueOfSubsingleton (Nonempty.some inferInstance)
+
 end Hom
 
+variable [WellFoundedLT J] {j : J} (iter₁ iter₂ iter₃ : Iteration ε j)
+
+/-- The canonical isomorphism between two objects in the category `Iteration ε j`. -/
+noncomputable def iso : iter₁ ≅ iter₂ where
+  hom := default
+  inv := default
+
+@[simp]
+lemma iso_refl : iso iter₁ iter₁ = Iso.refl _ := by aesop_cat
+
+lemma iso_trans : iso iter₁ iter₂ ≪≫ iso iter₂ iter₃ = iso iter₁ iter₃ := by aesop_cat
```

Comment:
I don't know how the place where you're applying `iso_trans` looks like, but would it make sense to provide `Unique (iter₁ ≅ iter₂)` instead (or in addition)?

Reply:
Thanks for the suggestion. I will try to see if this idea may help or not in the future downsteam PRs.